File: it.unibo.mbot2020/resources/usingCoap/basicrobotCoapClient.java
package usingCoap;
//
import org.eclipse.californium.core.CoapClient;
import org.eclipse.californium.core.CoapResponse;
import org.eclipse.californium.core.coap.MediaTypeRegistry;

public class basicrobotCoapClient {

private static CoapClient client = new CoapClient();

private static void sendToServer( String move ) {
   		String uriStr = "coap://192.168.1.8/"+move;
 		client.setURI( uriStr );
 		String arg = "{ \"activationCode\"  :  \"ACTIVATION_CODE\" }";
 		System.out.println("USING: " + uriStr);
 		CoapResponse res = client.put(arg, MediaTypeRegistry.TEXT_PLAIN);
 		System.out.println("RESPONSE=" + res.getResponseText() );
	}


private static void h() { sendToServer("h"); }
private static void w() { sendToServer("w"); }
private static void s() { sendToServer("s"); }
private static void r() { sendToServer("r"); }
private static void l() { sendToServer("l"); }
private static void x() { sendToServer("x"); }
private static void z() { sendToServer("z"); }

public static void main(String[] args) throws Exception {
	l();
	Thread.sleep(1000);
	r();
}

}

File: it.unibo.mbot2020/resources/usingCoap/basicrobotCoapClient.kt
package usingCoap

import org.eclipse.californium.core.CoapClient
import org.eclipse.californium.core.CoapResponse
import org.eclipse.californium.core.coap.MediaTypeRegistry
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.delay

object basicrobotCoapClient {

    private val client = CoapClient()

    private fun sendToServer(move: String) {
        val uriStr = "coap://192.168.1.8/$move"
        client.uri = uriStr
        val arg = "{ \"activationCode\"  :  \"ACTIVATION_CODE\" }"
        println("USING: $uriStr")
        val res = client.put(arg, MediaTypeRegistry.TEXT_PLAIN)
        println("RESPONSE=" + res.responseText)
    }

    fun h() {  sendToServer("h") }
    fun w() { sendToServer("w")  }
    fun s() { sendToServer("s")  }
    fun r() { sendToServer("r")  }
    fun l() { sendToServer("l")  }
    fun x() { sendToServer("x")  }
    fun z() { sendToServer("z")  }


}

    fun main( ) = runBlocking  {
        basicrobotCoapClient.w()
        delay(1000)
        basicrobotCoapClient.s()
        delay(1000)
        basicrobotCoapClient.h()
    }

